### PR TITLE
Fix invalid ruleset scope typings

### PR DESCRIPTION
**Pull request by cschanaj**
This PR makes compiling `scope` into `RegExp` automatic. Pass the string literal version of the `scope` parameter everywhere, except for when constructing rulesets/ ~WASM calls~. 

- `getScope` expected a required string parameter, but get none. It is also called multiple time when passed to `addFromJson`

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring existing code
- [ ] New Ruleset
- [ ] Existing Ruleset

### CODE REVIEW

File: chromium/background-scripts/rules.js [hunk 1]
```diff
@@ -232,26 +241,33 @@ RuleSets.prototype = {
...
   addFromJson: function(ruleJson, scope) {
-    const scope_obj = getScope(scope);
-
     if (this.wasm_rs) {
       this.wasm_rs.add_all_from_js_array(
         ruleJson,
         settings.enableMixedRulesets,
         this.ruleActiveStates,
-        scope);
+        getScope(scope));
```
Comment (Hainish) [inline chromium/background-scripts/rules.js:254]: Why change this string to a scope in JavaScript?  The scope will be cached when passed to Rust at any rate.  It looks like this passes a RegExp to the Rust bindings, which it isn't expecting.

**Review by Hainish — CHANGES_REQUESTED** [review]
See comment

File: chromium/background-scripts/rules.js (hunk 1 above)
Comment (cschanaj) [reply chromium/background-scripts/rules.js]: Look like a mistake made when I was trying to figure out the types of `scope` in different calls. The original source code uses string `scope` too. It should be a string. Changes made in f87434f

As the `getScope` function is being called in `Ruleset` constructor exclusively, I added 1c20e82 to reflect this. Only the internal representation of the `scope` is a `RegExp` now. Thanks.

**Comment by Hainish** [comment]
LGTM, thanks!

**Pull request merged**